crypto/asn1/bio_ndef.c

```diff
@@ -113,17 +113,16 @@ static int ndef_prefix(BIO *b, unsigned char **pbuf, int *plen, void *parg)
     ndef_aux = *(NDEF_SUPPORT **)parg;
 
     derlen = ASN1_item_ndef_i2d(ndef_aux->val, NULL, ndef_aux->it);
+    if (!*ndef_aux->boundary)
+        return 0;
```

Comment:
This would be better a little bit further up (before the `ndef_aux` assignment. Here is disturbs the "flow" of code where we calculate `derlen` and then allocate it. Also while we're at it lets make it `*ndef_aux->boundary == NULL`, which is more consistent with our preferred style these days.